### PR TITLE
ADEN-5918 Track FV as slot to kikimora

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/tracking/videoInfoTracker.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*global define, JSON*/
+define('ext.wikia.adEngine.tracking.videoInfoTracker',  [
+	'ext.wikia.adEngine.tracking.adInfoTracker'
+], function (tracker) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.tracking.videoInfoTracker';
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/tracking/videoInfoTracker.js]: MAJOR Remove the declaration of the unused 'logGroup' variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
MAJOR Remove this useless assignment to local variable "logGroup" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)

File: extensions/wikia/AdEngine/js/tracking/videoInfoTracker.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	function parseParameters(content, separator) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/tracking/videoInfoTracker.js]: MAJOR Remove the unused function parameter "separator". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/AdEngine/js/tracking/adInfoTracker.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+			'scroll_y': slotRegistry.getScrollY(slotName) || 0,
+			'rabbit': (rabbit && rabbit.getSerializedResults()) || ''
+		};
+
+		log(['prepareData', slotName, data], log.levels.debug, logGroup);
+
+		return data;
+	}
+
+	function track(slotName, pageParams, slotParams, adInfo, bidders) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/tracking/adInfoTracker.js:83]: MAJOR Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-js-rules%3AAmdExcludedFunctionParameterLimit)

File: extensions/wikia/AdEngine/js/tracking/adInfoTracker.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	'wikia.browserDetect',
+	'wikia.log',
+	'wikia.window',
+	require.optional('ext.wikia.adEngine.ml.rabbit')
+], function (adTracker, slotRegistry, browserDetect, log, win, rabbit) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.tracking.adInfoTracker';
+
+	function prepareData(slotName, pageParams, slotParams, creative, bidders) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/tracking/adInfoTracker.js]: MAJOR Function has a complexity of 34 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
MAJOR Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-js-rules%3AAmdExcludedFunctionParameterLimit)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 6 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/tracking/adInfoTracker.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+], function (adContext, adTracker, slotRegistry, browserDetect, log, win, rabbit) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.tracking.adInfoTracker';
+
+	function isEnabled() {
+		return adContext.getContext().opts.enableAdInfoLog;
+	}
+
+	function prepareData(slotName, pageParams, slotParams, creative, bidders) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/tracking/adInfoTracker.js]: MAJOR Function has a complexity of 34 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
MAJOR Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-js-rules%3AAmdExcludedFunctionParameterLimit)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/tracking/adInfoTracker.js (hunk 4 above)
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/tracking/adInfoTracker.js:14]: MAJOR Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-js-rules%3AAmdExcludedFunctionParameterLimit)
MAJOR Function has a complexity of 34 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: extensions/wikia/AdEngine/js/tracking/adInfoListener.js [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+			MOBILE_BOTTOM_LEADERBOARD: true,
+			MOBILE_IN_CONTENT: true,
+			MOBILE_PREFOOTER: true
+		};
+
+	function isEnabled() {
+		return adContext.getContext().opts.enableAdInfoLog;
+	}
+
+	function parseParameters(content) {
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/tracking/adInfoListener.js]: maybe we could use `querystring` module? It's pretty heavily tested by unittests and on production ;)

File: extensions/wikia/ArticleVideo/scripts/ooyala-player.js [hunk 7]
```diff
@@ -56,6 +58,10 @@ define('ooyala-player', [
 			self.onPlaybackReady();
 		});
 
+		messageBus.subscribe(OO.EVENTS.ADS_PLAYED, 'video-tracker', function () {
+			self.adIndex += 1;
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/scripts/ooyala-player.js:62]: As an alternative, you can pass right context to the callback method using `.bind(this)`. Having that said `self` is already used in this file so it's ok, to leave it as it is ;)

File: extensions/wikia/ArticleVideo/scripts/ooyala-player.js [hunk 8]
```diff
@@ -186,10 +193,35 @@ define('ooyala-player', [
...
 
-						options.adTrackingParams.lineItemId = adData.adId;
-						options.adTrackingParams.creativeId = adData.creativeId;
+						if (currentAd) {
+							wrapperId = currentAd.getWrapperAdIds();
+							if (wrapperId.length) {
+								options.adTrackingParams.lineItemId = wrapperId[0];
+							}
+
+							wrapperId = currentAd.getWrapperCreativeIds();
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/scripts/ooyala-player.js]: can we use separate variable for `creativeId`?

File: extensions/wikia/ArticleVideo/scripts/ooyala-player.js [hunk 9]
```diff
@@ -186,10 +193,35 @@ define('ooyala-player', [
...
+							options.adTrackingParams.lineItemId = adData.adId;
+							options.adTrackingParams.creativeId = adData.creativeId;
+						}
 
-						options.adTrackingParams.lineItemId = adData.adId;
-						options.adTrackingParams.creativeId = adData.creativeId;
+						if (currentAd) {
+							wrapperId = currentAd.getWrapperAdIds();
+							if (wrapperId.length) {
+								options.adTrackingParams.lineItemId = wrapperId[0];
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/scripts/ooyala-player.js]: Good catch with wrapper Ids! :+1:

File: extensions/wikia/AdEngine/js/tracking/adInfoListener.js (hunk 6 above)
Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/tracking/adInfoListener.js]: Absolutely! Thanks!

File: extensions/wikia/ArticleVideo/scripts/ooyala-player.js (hunk 8 above)
Comment (fraszczakszymon) [reply extensions/wikia/ArticleVideo/scripts/ooyala-player.js]: Done 👍

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**